tests: Use options and fixtures to control all browsers

jsfehler opened the pull request:
This PR refactors the tests to:

  - Remove the class based-structure and replace it with browser flags and isolated fixtures.
  - Replace an autouse fixture for visiting the dummy site with an explicit call in each test. This makes it easier to understand each test in isolation.
  - Run more tests for each browser.
  - Use a standard structure for xfail and skip instead of the macos tests subclassing tests.
  - Use a fixture to get the dummy app's URL instead of importing from it's internals.
  - Improve test organization by splitting them into more files and directories based on what they do.
  - Remove unnecessary imports from the requirements file for the tests


Before vs This PR:

- Firefox: 446  vs 504
- Chrome: 443 vs 498
- Edge: 443 vs 498
- Remote Chrome & Firefox: 380 + 2 skip vs 496
- Remote Safari: 139 + 1 skip + 14 xfail + 41 error vs 229 + 3 skip + 13 xfail + 3 xpass
 - Django + flask + zope: 498 + 7 skip vs 499

fsouza reviewed (approved):
woah this is large, but generally looks good. I'm approving after skimming through the code and given that CI is green 😁

jsfehler commented:
> woah this is large, but generally looks good. I'm approving after skimming through the code and given that CI is green 😁

Yeah, sorry about that. I did as much as I could over separate PRs but getting it done in one shot made addressing issues a lot easier without putting in temporary hacks.
